analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SillyMathematicalComparison.cs

```diff
@@ -0,0 +1,413 @@
+            _ = veryBig > ul; // Noncompliant {{Comparison to this constant is useless; the constant is outside the range of type 'ulong'}}
+            _ = ul > verySmall; // Noncompliant {{Comparison to this constant is useless; the constant is outside the range of type 'ulong'}}
+            _ = verySmall < ul; // Noncompliant {{Comparison to this constant is useless; the constant is outside the range of type 'ulong'}}
+            _ = verySmall == ul; // Noncompliant {{Comparison to this constant is useless; the constant is outside the range of type 'ulong'}}
+            _ = ul == veryBig; // Noncompliant {{Comparison to this constant is useless; the constant is outside the range of type 'ulong'}}
+        }
+
+        public void Floats()
+        {
+            const float smallFloat = float.MinValue;
+            const double smallDouble = float.MinValue;
+
+            const float bigFloat = float.MaxValue;
+            const double bigDouble = float.MaxValue;
+
+            System.Single f = GetValue<float>();
+
+            _ = bigFloat <= f; // Compliant, not (always true) or (always false)
+            _ = bigDouble <= f; // Compliant, not (always true) or (always false)
+
+            _ = smallFloat >= f; // Compliant, not (always true) or (always false)
+            _ = smallDouble >= f; // Compliant, not (always true) or (always false)
+
+            const double veryBig = double.MaxValue;
+            const double verySmall = double.MinValue;
+
+            _ = f < veryBig; // Noncompliant {{Comparison to this constant is useless; the constant is outside the range of type 'float'}}
+            _ = veryBig > f; // Noncompliant {{Comparison to this constant is useless; the constant is outside the range of type 'float'}}
+            _ = f > verySmall; // Noncompliant {{Comparison to this constant is useless; the constant is outside the range of type 'float'}}
+            _ = verySmall < f; // Noncompliant {{Comparison to this constant is useless; the constant is outside the range of type 'float'}}
+            _ = verySmall == f; // Noncompliant {{Comparison to this constant is useless; the constant is outside the range of type 'float'}}
+            _ = f == veryBig; // Noncompliant {{Comparison to this constant is useless; the constant is outside the range of type 'float'}}
+        }
+
+        public void ConstantInBothOperands()
+        {
+            _ = "Happy" == "Sad"; // Noncompliant {{Don't compare constant values}}
+            _ = "Ipsum" != "Lorem"; // Noncompliant {{Don't compare constant values}}
+
+            var george = true;
+            var boole = false;
+
+            _ = george == boole; // Noncompliant {{Don't compare constant values}}
+
+            var number1 = 128;
+            var number2 = 42;
+
+            _ = 42 == 256; // Noncompliant {{Don't compare constant values}}
+            _ = number1 >= 256; // Noncompliant {{Don't compare constant values}}
```

Comment:
Why is `number1` recognized as a constant? It actually isn't.

Reply:
Fixed with the usage of the default `GetConstantValue` instead of our extension method.